diff --git a/node_modules/@bookshop/astro-engine/lib/engine.js b/node_modules/@bookshop/astro-engine/lib/engine.js
index 91b7ca7..8352ad2 100644
--- a/node_modules/@bookshop/astro-engine/lib/engine.js
+++ b/node_modules/@bookshop/astro-engine/lib/engine.js
@@ -151,6 +151,12 @@ export class Engine {
         hasRenderedHead: true,
         hasDirectives: new Set(),
       },
+      clientDirectives: new Map([
+       	["load", "bookshop-placeholder"],
+       	["idle", "bookshop-placeholder"],
+       	["visible", "bookshop-placeholder"],
+        ["media", "bookshop-placeholder"],
+      ]),
       slots: null,
       props,
       createAstro(astroGlobal, props, slots) {
