diff --git a/node_modules/@bookshop/astro-engine/lib/modules/image.astro b/node_modules/@bookshop/astro-engine/lib/modules/image.astro
new file mode 100644
index 0000000..0b9b8cb
--- /dev/null
+++ b/node_modules/@bookshop/astro-engine/lib/modules/image.astro
@@ -0,0 +1,33 @@
+---
+const props = Astro.props;
+
+if (props.alt === undefined || props.alt === null) {
+  throw new Error("Image missing 'alt'");
+}
+
+// As a convenience, allow width and height to be string with a number in them, to match HTML's native `img`.
+if (typeof props.width === "string") {
+  props.width = parseInt(props.width);
+}
+
+if (typeof props.height === "string") {
+  props.height = parseInt(props.height);
+}
+
+if (!props || typeof props !== "object") {
+  throw new Error("Image missing options");
+}
+if (typeof props.src === "undefined") {
+  throw new Error("Image missing 'src'");
+}
+
+const resolvedSrc =
+  typeof props.src === "object" && "then" in props.src
+    ? (await props.src).default ?? (await props.src)
+    : props.src;
+
+const src = typeof resolvedSrc === "object" ? resolvedSrc.src : resolvedSrc;
+delete props.src;
+---
+
+<img src={src} {...props} />
diff --git a/node_modules/@bookshop/astro-engine/lib/modules/picture.astro b/node_modules/@bookshop/astro-engine/lib/modules/picture.astro
new file mode 100644
index 0000000..20d1519
--- /dev/null
+++ b/node_modules/@bookshop/astro-engine/lib/modules/picture.astro
@@ -0,0 +1,7 @@
+---
+import Image from "./image.astro";
+---
+
+<picture>
+  <Image {...Astro.props} />
+</picture>
