diff --git a/node_modules/@bookshop/astro-engine/lib/engine.js b/node_modules/@bookshop/astro-engine/lib/engine.js
index 91b7ca7..64a53c1 100644
--- a/node_modules/@bookshop/astro-engine/lib/engine.js
+++ b/node_modules/@bookshop/astro-engine/lib/engine.js
@@ -151,6 +151,7 @@ export class Engine {
         hasRenderedHead: true,
         hasDirectives: new Set(),
       },
+      clientDirectives: new Map(),
       slots: null,
       props,
       createAstro(astroGlobal, props, slots) {
