diff --git a/node_modules/astro-remote/lib/utils.ts b/node_modules/astro-remote/lib/utils.ts
index b9d9273..6cc4d6e 100644
--- a/node_modules/astro-remote/lib/utils.ts
+++ b/node_modules/astro-remote/lib/utils.ts
@@ -1,7 +1,9 @@
 import { marked } from 'marked';
 import { transform } from 'ultrahtml';
+import swap from 'ultrahtml/transformers/swap';
+import sanitize from 'ultrahtml/transformers/sanitize';
 import { jsx as h } from 'astro/jsx-runtime';
-import { renderJSX } from 'astro/runtime/server/jsx';
+import { renderJSX } from 'astro/runtime/server/jsx.js';
 import { __unsafeHTML } from 'ultrahtml';
 import * as entities from "entities";
 
@@ -96,18 +98,18 @@ export async function markdown(
   })
   const content = await marked.parse(dedent(input));
 
-  return transform(content, {
-    sanitize: opts.sanitize,
-    components: opts.components,
-  });
+  return transform(content, [
+    swap(opts.components),
+    sanitize(opts.sanitize),
+  ]);
 }
 
 export async function html(
   input: string,
   opts: HTMLOptions = {}
 ): Promise<string> {
-  return transform(dedent(input), {
-    sanitize: opts.sanitize,
-    components: opts.components,
-  });
+  return transform(dedent(input), [
+    swap(opts.components),
+    sanitize(opts.sanitize),
+  ]);
 }
